src/Notes/DeprecatedNotes.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * Define deprecated classes to support changing the naming convention of
+ * admin notes.
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+
+/**
+ * WC_Admin_Note (deprecated, use Note instead.
+ */
+class WC_Admin_Note extends Note {}
```

Comment:
It would be neat to have this file use [`_deprecated_function()`](https://developer.wordpress.org/reference/functions/_deprecated_function/) to highlight the deprecations more to developers, but I admit handling all cases might need code that's a little too clever for it's own good.

A rough sketch:

- Create a `Deprecated_Class` class that expects a `$new_classname` property and implements PHP's magic [`__call()` and `__callStatic()`](https://www.php.net/manual/en/language.oop5.overloading.php#object.call) to intercept calls in order to warn with `_deprecated_function()` before passing the call on to the `$new_classname`
- Define `Deprecated_Class` instances for each class being replaced, setting the `$new_classname` property appropriately

```php
class WC_Admin_Notes_Choose_Niche extends DeprecatedClass {
    $new_classname = 'Choose_Niche';
}
```

Reply:
Very nice. Done in e6ea2031f670458c36c2dce4b574b802ad00ef95. Thanks :-)